### PR TITLE
#1 Add solution with validation library

### LINKED ISSUE #1: Add solution with Functional Validation shared library
**As a** developer,
**I want** a Functional Validation shared library with tests project inside solution that has pipelines for build and tests check in pull requests and publishing library as package after merging to develop
**so that** I have a template for further development.

### PR DESCRIPTION
**Pull request by krzysztofzajaczkowski**
closes #1

### CODE REVIEW

File: src/shared/FunctionalValidation.Tests/FakeTests.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿using CSharpFunctionalExtensions;
+using Machine.Fakes;
+using Machine.Specifications;
+using Result = CSharpFunctionalExtensions.Result;
+
+namespace FunctionalValidation.Tests;
+
+// ReSharper disable InconsistentNaming
+// ReSharper disable ArrangeTypeMemberModifiers
```
Comment (RafKulas) [inline src/shared/FunctionalValidation.Tests/FakeTests.cs:9]: I assume it is just for resharper to not scream about some warnings. I think you should consult with @bubus128 that both of you want to use resharper

File: .github/workflows/functional-validation-publish.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+name: FunctionalValidation-PublishPackage
```
Comment (RafKulas) [inline .github/workflows/functional-validation-publish.yml:1]: Why is it a `FunctionalValidation`, not just `Validation`? IMO `FunctionalValidation` sounds like Functional/API Tests

File: .github/workflows/functional-validation-publish.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+jobs:
+  publish:
+    if: ${{ github.event.pull_request.merged }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: 6.0.x
```
Comment (RafKulas) [inline .github/workflows/functional-validation-publish.yml]: Can we keep it as GitHub var? I think there would be more `dotnet-version: x.y.z`

**Review by RafKulas — APPROVED** [review]

**Review by RafKulas — APPROVED** [review]

File: src/shared/FunctionalValidation.Tests/FakeTests.cs (hunk 1 above)
Comment (krzysztofzajaczkowski) [reply src/shared/FunctionalValidation.Tests/FakeTests.cs:9]: I think it doesn't matter if @bubus128 wants to use ReSharper. If not, that's okay. He doesn't have to insert these comments then. I'll add them whenever I will be making changes to test classes without these. Moreover, he'll probably use different library for unit tests, one that wouldn't need these comments either way.

File: .github/workflows/functional-validation-publish.yml (hunk 2 above)
Comment (krzysztofzajaczkowski) [reply .github/workflows/functional-validation-publish.yml:1]: Haven't had any original idea for a library name. The point is that this validation library will use functional programming constructs from `CSharpFunctionalExtensions` (`Result<T>` or `Result<T,E>`).

File: .github/workflows/functional-validation-publish.yml (hunk 3 above)
Comment (krzysztofzajaczkowski) [reply .github/workflows/functional-validation-publish.yml]: I've created local GitHub action that loads variables from env files in `.github/vars` directory.

**Review by RafKulas — APPROVED** [review]

**Pull request merged**